pandas-stubs/plotting/_core.pyi

```diff
@@ -1,94 +1,250 @@
+    ax: Axes | None = ...,
+    fontsize: float | str | None = ...,
+    rot: float = ...,
     grid: bool = ...,
-    xlabelsize=...,
-    xrot=...,
-    ylabelsize=...,
-    yrot=...,
-    ax=...,
-    sharex: bool = ...,
-    sharey: bool = ...,
-    figsize=...,
-    layout=...,
-    bins: int = ...,
-    backend=...,
+    figsize: tuple[float, float] | None = ...,
+    layout: tuple[int, int] | None = ...,
+    return_type: Literal["axes"] | None = ...,
     **kwargs,
-): ...
+) -> Axes: ...
+@overload
 def boxplot(
     data: DataFrame,
-    column: str | list[str] | None = ...,
-    by: str | list[str] | None = ...,
-    ax: PlotAxes | None = ...,
+    column: Hashable | list[HashableT] | None = ...,
+    by: Hashable | list[HashableT] | None = ...,
+    ax: Axes | None = ...,
     fontsize: float | str | None = ...,
     rot: float = ...,
     grid: bool = ...,
     figsize: tuple[float, float] | None = ...,
     layout: tuple[int, int] | None = ...,
-    return_type: str | None = ...,
-): ...
-def boxplot_frame(
-    self,
-    column=...,
-    by=...,
-    ax=...,
-    fontsize=...,
-    rot: int = ...,
-    grid: bool = ...,
-    figsize=...,
-    layout=...,
-    return_type=...,
-    backend=...,
+    *,
+    return_type: Literal["dict"],
     **kwargs,
-): ...
-def boxplot_frame_groupby(
-    grouped,
-    subplots: bool = ...,
-    column=...,
-    fontsize=...,
-    rot: int = ...,
+) -> dict[str, list[Line2D]]: ...
+@overload
+def boxplot(
+    data: DataFrame,
+    column: Hashable | list[HashableT] | None = ...,
+    by: Hashable | list[HashableT] | None = ...,
+    ax: Axes | None = ...,
+    fontsize: float | str | None = ...,
+    rot: float = ...,
     grid: bool = ...,
-    ax=...,
-    figsize=...,
-    layout=...,
-    sharex: bool = ...,
-    sharey: bool = ...,
-    backend=...,
+    figsize: tuple[float, float] | None = ...,
+    layout: tuple[int, int] | None = ...,
+    *,
+    return_type: Literal["both"],
     **kwargs,
-): ...
+) -> _BoxPlotT: ...
 
 class PlotAccessor(PandasObject):
     def __init__(self, data) -> None: ...
     def __call__(self, *args, **kwargs): ...
-    def line(self, x=..., y=..., **kwargs) -> PlotAccessor: ...
-    def bar(self, x=..., y=..., **kwargs) -> PlotAccessor: ...
-    def barh(self, x=..., y=..., **kwargs) -> PlotAccessor: ...
-    def box(self, by=..., **kwargs) -> PlotAccessor: ...
-    def hist(self, by=..., bins: int = ..., **kwargs) -> PlotAccessor: ...
-    def kde(self, bw_method=..., ind=..., **kwargs) -> PlotAccessor: ...
-    density = ...
-    def area(self, x=..., y=..., **kwargs) -> PlotAccessor: ...
-    def pie(self, **kwargs) -> PlotAccessor: ...
-    def scatter(self, x, y, s=..., c=..., **kwargs) -> PlotAccessor: ...
+    @overload
+    def line(
+        self,
+        x: Hashable | None = ...,
+        y: Hashable | None = ...,
+        color: _PlotAccessorColor = ...,
+        subplots: Literal[False] | None = ...,
+        **kwargs,
```

Comment:
All of these functions share the same `kwargs` defined in the docs for `DataFrame.plot()`, so I think we should include them in each of these stubs (unfortunately, lots of repeated code) and then not have `**kwargs`.

Reply:
Not sure we can get rid of kwargs since these are passed through to matplotlib, and some are plot specific.

Reply:
> Not sure we can get rid of kwargs since these are passed through to matplotlib, and some are plot specific.

OK, but we can at least add all the arguments that are documented in `DataFrame.plot()`

Reply:
I've added these and tested them.